### PR TITLE
added: implement getBoundaryNodes for ASMu2D and ASMu3D

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by kmokstad** [comment]
Looks good, but need to test this one also wrt. boundary/reaction force calculations before merging.
And do we support 3D LR yet?

File: src/ASM/LR/ASMu2D.C [hunk 1]
```diff
@@ -1710,7 +1710,37 @@ bool ASMu2D::evalSolution (Matrix& sField, const IntegrandBase& integrand,
...
+  LR::parameterEdge edge;
+  switch (lIndex) {
+  case 1: edge = LR::WEST; break;
+  case 2: edge = LR::EAST; break;
+  case 3: edge = LR::SOUTH; break;
+  case 4: edge = LR::NORTH; break;
+  default: return;
+  }
+
+  if (!this->getBasis(basis)) return; // silently ignore empty patches
```
Comment (kmokstad) [inline src/ASM/LR/ASMu2D.C]: Kan med fordel flyttes opp til forman switch-statementet, som vil være vasted i tilfelle return her.

**Comment by akva2** [comment]
3D code added, things moved around and simplified by using the existing getEdgeNodes() helper function.

File: src/ASM/LR/ASMu3D.C [hunk 2]
```diff
@@ -1917,9 +1917,50 @@ bool ASMu3D::evalSolution (Matrix& sField, const IntegrandBase& integrand,
...
+  case 2: edge = LR::EAST; break;
+  case 3: edge = LR::SOUTH; break;
+  case 4: edge = LR::NORTH; break;
+  case 5: edge = LR::BOTTOM; break;
+  case 6: edge = LR::TOP; break;
+  default: return;
+  }
+
+  nodes = this->getEdgeNodes(edge, basis);
+
```
Comment (kmokstad) [inline src/ASM/LR/ASMu3D.C]: Randa av et 3D element bastår av Faces og ikke Edges, så denne metoden burde vel hete getFaceNodes (og ikke getEdgeNodes) for å unngå forvirring, eller er det bare funksjonene som ligger lags en kant som returneres. Der er i såfall feil funksjonalitet.

Comment (akva2) [inline src/ASM/LR/ASMu3D.C]: c&p. det skal være getFaceNodes.

Comment (akva2) [inline src/ASM/LR/ASMu3D.C]: fikset (og squashet).

**Comment by VikingScientist** [comment]
Looked through this, and it all looks good :+1: 

@kmokstad : We do progressively support more and more volumetric LR. There is a few commits with respect to this on my IFEM fork which I did before my paternety leave at https://github.com/VikingScientist/IFEM-2/commits/master . I can't seem to find these merged into trunk, so I'll have a look into brushing dust of this one

**Comment by kmokstad** [comment]
Merged manually.

**Pull request closed without merging**